pkg/instance/instance.go

```diff
@@ -59,3 +80,29 @@ func (i *Info) UpgradeRequired() bool {
 	// fully deconfigured before being configured by the current version.
 	return true
 }
+
+// GetIPv4Address returns an IPv4 address of the instance. An error will be thrown if the instance object has
+// an IPv6 address or a DNS address that does not resolve to an IPv4 network address.
+func (i *Info) GetIPv4Address() (string, error) {
+	if ip := net.ParseIP(i.Address); ip != nil {
+		// Address is either an IPv6 or IPv4 address
+		ipv4 := ip.To4()
+		if ipv4 == nil {
+			return "", newNoIPv4Err(errors.Errorf("error using IP %s: only IPv4 addresses are supported", ip.String()))
```

Comment:
Would the final error read `No IPv4 address found: error using IP: only IPv4 addresses are supported`?

Reply:
I will make the inner error more clear and less repetitive: `unable to use IPv6 address <ip>`

Reply:
But that assumes the error is always because an IPv6 address is present. Could there be other reasons?